Purchases/Public/EntitlementInfo.swift

```diff
@@ -246,21 +246,21 @@ import Foundation
                   entitlementData: entitlementData,
                   productData: productData,
                   requestDate: requestDate,
-                  dateFormatter: .iso8601SecondsDateFormatter,
+                  dateFormatter: ISO8601DateFormatter.default,
```

Comment:
i know this isn't part of this PR...but looks like we only use the dateDecodingStrategy here, should we just pass that?

Reply:
Oh that's a good idea, let me check that.

Reply:
Great find! I just refactored this by passing a strategy instead so the protocol is simpler. Thanks!